src/main/java/uk/gov/justice/laa/ccms/service/DecisionReportTransformation.java

```diff
@@ -402,35 +403,86 @@ private DecisionReportType getDecisionReport(GlobalInstanceType globalInstanceTy
     }
     return null;
   }
-
+  
+  /**
+   * OPA10 global entity isn't supported in OPA18 as the AssessService hierarchy is different
+   * @param globalEntityId
+   * @param assess10Response
+   */
   public void restructureDecisionReport( String globalEntityId,
       com.oracle.determinations.server._10_0.rulebase.types.AssessResponse assess10Response ){
-
+    
+    //Reset at GOAL attribute DecisionReport
     DecisionReport decisionReport = getDecisionReport(assess10Response);
-
     if ( decisionReport != null ) {
       //Loop through nodes in Decision Report & retrieve AttributeNodeTypes and RelationshipNodeTypes
       List<?> nodes = (List<?>) decisionReport.getRelationshipDecisionNodeOrAttributeDecisionNode();
       resetGlobalEntityId(nodes, globalEntityId);
     }
+    
+    //Reset at warning level
+    DecisionReport warningDecisionReport = getErrorOrWarningDecisionReport(globalEntityId, assess10Response, "warning");
+    if ( warningDecisionReport != null ) {
+      //Loop through nodes in Decision Report & retrieve AttributeNodeTypes and RelationshipNodeTypes
+      List<?> nodes = (List<?>) warningDecisionReport.getRelationshipDecisionNodeOrAttributeDecisionNode();
+      resetGlobalEntityId(nodes, globalEntityId);
+    }
   }
-
+  
+  /**
+   * Set globalId at warning if entity is global & return warning DecisionReport
+   * @param globalEntityId
+   * @param assess10Response
+   * @return
+   */
+  private DecisionReport getErrorOrWarningDecisionReport(String globalEntityId, AssessResponse assess10Response, String eventName) {
+    logger.debug("eventName ==== " + eventName);
+    if ( (assess10Response.getEvents() != null) && (assess10Response.getEvents().getEvent() != null) ) {
+      logger.debug("============= 1");
+      for ( RulebaseEvent event : assess10Response.getEvents().getEvent() ) {
+        logger.debug("============= event.getName() = " + event.getName());
+        if ( eventName.equalsIgnoreCase(event.getName())) {
+          logger.debug("============= 2");
+          if ( "global".equalsIgnoreCase(event.getEntityId()) ) {
+            event.setEntityId(globalEntityId);
+          }
+          logger.debug("============= 3");
+          return event.getDecisionReport();
+        }
+      }
+    }
+    logger.debug("============= 4");
```

Comment:
I think I'd find these numbers in the log messages a bit confusing, but happy to leave them in if they're helpful to you 👍

Reply:
These were added to investigate different types of payloads (Billing & Means) where response types vary. This will be removed after completion of BulkClaim changes